perfkitbenchmarker/linux_benchmarks/dpb_wordcount_benchmark.py

```diff
@@ -0,0 +1,140 @@
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Runs the word count job on data processing backends.
+
+WordCount example reads text files and counts how often words occur. The input
+is text files and the output is text files, each line of which contains a word
+and the count of how often it occured, separated by a tab.
+"""
+
+import tempfile
+
+from perfkitbenchmarker import configs
+from perfkitbenchmarker.dpb_service import BaseDpbService
+from perfkitbenchmarker import flags
+from perfkitbenchmarker.providers.aws import aws_dpb_emr
+from perfkitbenchmarker.providers.gcp import gcp_dpb_dataproc
+from perfkitbenchmarker.providers.gcp import gcp_dpb_dataflow
+
+BENCHMARK_NAME = 'dpb_wordcount_benchmark'
+
+BENCHMARK_CONFIG = """
+dpb_wordcount_benchmark:
+  description: Run word count on dataflow and dataproc
+  flags:
+    dpb_wordcount_out_fs: gs
+    dpb_wordcount_out_base: saksena-df
+  dpb_service:
+    service_type: dataproc
+    worker_group:
+      vm_spec:
+        GCP:
+          machine_type: n1-standard-1
+          boot_disk_size: 500
+        AWS:
+          machine_type: m1.medium
+      disk_spec:
+        GCP:
+          disk_type: nodisk
+        AWS:
+          disk_size: 500
+          disk_type: gp2
+    worker_count: 2
+"""
+
+flags.DEFINE_string('dpb_wordcount_gcs_input',
+                    'gs://dataflow-samples/shakespeare/kinglear.txt',
+                    'Input for word count')
+flags.DEFINE_string('dpb_wordcount_s3_input',
+                    's3://pkb-shakespeare/kinglear.txt',
+                    'Input for word count')
+flags.DEFINE_enum('dpb_wordcount_out_fs', BaseDpbService.HDFS_OUTPUT_FS,
+                  [BaseDpbService.HDFS_OUTPUT_FS, BaseDpbService.GCS_OUTPUT_FS,
+                   BaseDpbService.S3_OUTPUT_FS],
+                  'File System to use for the job output')
+flags.DEFINE_string('dpb_wordcount_out_base', None,
+                    'Base directory for word count output')
+
+FLAGS = flags.FLAGS
+
+
+def GetConfig(user_config):
+    return configs.LoadConfig(BENCHMARK_CONFIG, user_config, BENCHMARK_NAME)
+
+
+def Prepare(benchmark_spec):
+    pass
+
+
+def Run(benchmark_spec):
+    # Get handle to the dpb service
+    dpb_service = benchmark_spec.dpb_service
+
+    """
+    Create a file handle to contain the response from running the job on
+    the dpb service
+    """
+    stdout_file = tempfile.NamedTemporaryFile(suffix='.stdout',
+                                              prefix='spark_benchmark',
+                                              delete=False)
+    stdout_file.close()
+
+    # Switch the parameters for submit job function of specific dpb service
+    job_arguments = []
+    if dpb_service.SERVICE_TYPE == 'dataproc':
```

Comment:
I have refactored the common elements. The dataflow has some additional parameters that are set, and I don't have a very clean way of refactoring that further. PTLA and let me know if you there are any additional suggestions.